webhookURLValidator_test.go

```diff
@@ -0,0 +1,473 @@
+			assert.NoError(err)
+			res := RejectHosts(tc.rejectHosts)(u)
+			assert.True(errors.Is(res, tc.expectedErr),
+				fmt.Errorf("error [%v] doesn't contain error [%v] in its err chain",
+					res, tc.expectedErr),
+			)
+		})
+	}
+}
+
+func TestRejectAllIPs(t *testing.T) {
+	tcs := []struct {
+		desc        string
+		url         string
+		expectedErr error
+	}{
+
+		{
+			desc: "Non-IP Success",
+			url:  "http://www.example.com:1030/software/index.html",
+		},
+		{
+			desc:        "Loopback IP with Port Failure",
+			url:         "https://127.0.0.1:1030",
+			expectedErr: errIPGivenAsHost,
+		},
+		{
+			desc:        "Loopback IP with no port Failure",
+			url:         "https://127.0.0.1",
+			expectedErr: errIPGivenAsHost,
+		},
+		{
+			desc: "No Host Success",
+			url:  "/example/test",
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.desc, func(t *testing.T) {
+			assert := assert.New(t)
+			u, err := url.ParseRequestURI(tc.url)
+			assert.NoError(err)
+			res := RejectAllIPs()(u)
+			assert.True(errors.Is(res, tc.expectedErr),
+				fmt.Errorf("error [%v] doesn't contain error [%v] in its err chain",
+					res, tc.expectedErr),
+			)
+		})
+	}
+}
+
+func TestRejectLoopback(t *testing.T) {
+	tcs := []struct {
+		desc        string
+		url         string
+		expectedErr error
+	}{
+		{
+			desc: "Non loopback URL Success",
+			url:  "http://www.example.com:1030/software/index.html",
+		},
+		{
+			desc:        "Loopback IP with Port Failure",
+			url:         "https://127.0.0.1:1030",
+			expectedErr: errLoopbackGivenAsHost,
+		},
+		{
+			desc:        "Loopback URL with Port Failure",
+			url:         "https://localhost:9000",
+			expectedErr: errLoopbackGivenAsHost,
+		},
+		{
+			desc: "IP Host Success",
+			url:  "http://96.118.133.128",
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.desc, func(t *testing.T) {
+			assert := assert.New(t)
+			u, err := url.ParseRequestURI(tc.url)
+			assert.NoError(err)
+			res := RejectLoopback()(u)
+			assert.True(errors.Is(res, tc.expectedErr),
+				fmt.Errorf("error [%v] doesn't contain error [%v] in its err chain",
+					res, tc.expectedErr),
+			)
+		})
+	}
+}
+
+func TestInvalidSubnets(t *testing.T) {
+	tcs := []struct {
+		desc               string
+		url                string
+		subnetsList        []string
+		expectedInitialErr error
+		expectedLatterErr  error
+	}{
+		{
+			desc:               "Invalid subnet provided Failure",
+			url:                "https://2001:db8:a0b:12f0::1/32",
+			subnetsList:        []string{"2001:db8:a0b:12f0::1//32"},
+			expectedInitialErr: errInvalidSubnet,
+		},
+		{
+			desc:              "IP in subnet Failure",
+			url:               "https://192.0.2.56",
+			subnetsList:       []string{"192.0.2.1/24"},
+			expectedLatterErr: errIPinInvalidSubnets,
+		},
+		{
+			desc: "Nil subnet Success",
+			url:  "https://192.0.2.56:1030",
+		},
+		{
+			desc:        "Valid IP given with valid subnets Success",
+			url:         "https://2001:db8:a0b:12f0::1/32",
```

Comment:
Is this a valid URL? I thought the ipv6 format would be something like: `https://[ipv6 address]:8090/profit`